### PR TITLE
br: split in each TiKV and wait scatter at last

### LINKED ISSUE #50665: br restore reported split region error when setting granularity=coarse-grained
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. restore point with  --granularity=coarse-grained
2. restore 2nd restore with checkpoints

### 2. What did you expect to see? (Required)
pass
### 3. What did you see instead (Required)
```
Error: split region failed: err=message:"key 748000000000009FFF3F5F698000000000FF000001FB00000000FB is not in region key range [748000000000009EFF6A5F72FFFFFFFFFFFFFFFFFF0000000000FB, 748000000000009FFF3F5F698000000000FF000001FB00000000FB) for region 283948" key_not_in_region:<key:"t\200\000\000\000\000\000\237\377?_i\200\000\000\000\000\377\000\000\001\373\000\000\000\000\373" region_id:283948 start_key:"t\200\000\000\000\000\000\236\377j_r\377\377\377\377\377\377\377\377\377\000\000\000\000\000\373" end_key:"t\200\000\000\000\000\000\237\377?_i\200\000\000\000\000\377\000\000\001\373\000\000\000\000\373" > 
....
```
### 4. What is your TiDB version? (Required)
sh-5.1# /br --version
Release Version: v7.6.0
Git Commit Hash: 370b2d59f84cc680a6f7e74a83253338fb41dcc4
Git Branch: heads/refs/tags/v7.6.0
Go Version: go1.21.5
UTC Build Time: 2024-01-21 13:28:35
Race Enabled: false

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #50665 

Problem Summary:
the new split & scatter method has some bugs with checkpoint restore.
### What changed and how does it work?
We found we can use the original function to implement it, so remove these unstable functinos.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/split.go [hunk 1]
```diff
@@ -125,122 +124,47 @@ func (rs *RegionSplitter) ExecuteSplit(
...
+	return rs.executeSplitByRanges(ctx, sctx, scanStartKey, sortedKeys, granularity)
 }
 
 func (rs *RegionSplitter) executeSplitByRanges(
 	ctx context.Context,
 	splitContext SplitContext,
-	sortedRanges []rtree.Range,
+	scanStartKey []byte,
+	sortedKeys [][]byte,
+	granularity string,
```
[the remark this answers is not in the document]
Comment (Leavrth) [reply br/pkg/restore/split.go:135]: yes

**Review by 3pointer — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**